### PR TITLE
Feature/new image

### PR DESCRIPTION
**Pull request by Amirjf**

### CODE REVIEW

File: src/components/button/content/button_types.d.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import * as React from 'react';
 
 export type ButtonProps = {
-  children: React.ReactNode;
+  children?: React.ReactNode;
```
Comment (alirezatahani) [inline src/components/button/content/button_types.d.ts:4]: chera ? 
children bara btn ejbarie

File: src/components/button/content/button_types.d.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
   style?: React.CSSProperties;
   size?: 'sm' | 'md' | 'lg';
   variant?: 'outlined' | 'text' | 'filled';
   color?: 'primary' | 'secondary' | string;
   rounded?: boolean;
   block?: boolean;
   disabled?: boolean;
+  value?: string | number;
```
Comment (alirezatahani) [inline src/components/button/content/button_types.d.ts:12]: value ? btn mage value dare ?

File: src/components/button/content/button_types.d.ts [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+  onClick?: () => void;
+  getProps?: any;
```
Comment (alirezatahani) [inline src/components/button/content/button_types.d.ts:14]: ino dg pakesh kon

File: src/components/image/content/Image.tsx [hunk 4]
```diff
@@ -1,20 +1,24 @@
 import * as React from 'react';
-import { StyleImage } from '../styles/image.styles';
+import { ImageContainer, StyledImage } from '../styles/image.styles';
 import { ImageProps } from './image_types';
 
 const Image: React.FC<ImageProps> = ({ src, ...props }: ImageProps) => {
+  const [imgSrc, setImgSrc] = React.useState<string | undefined>(src);
```
Comment (alirezatahani) [inline src/components/image/content/Image.tsx:6]: hichvaght az props mostaghim naro too state in ghalate

**Review by alirezatahani0 — CHANGES_REQUESTED** [review]

File: src/components/button/content/button_types.d.ts (hunk 3 above)
Comment (Amirjf) [reply src/components/button/content/button_types.d.ts]: done

File: src/components/button/content/button_types.d.ts (hunk 2 above)
Comment (Amirjf) [reply src/components/button/content/button_types.d.ts]: na :` )

File: src/components/button/content/button_types.d.ts (hunk 1 above)
Comment (Amirjf) [reply src/components/button/content/button_types.d.ts]: fixed

File: src/components/image/content/Image.tsx (hunk 4 above)
Comment (Amirjf) [reply src/components/image/content/Image.tsx]: i've added onError to types

File: src/components/image/content/Image.tsx [hunk 5]
```diff
@@ -1,20 +1,18 @@
...
-      <StyleImage src={src} {...props} />
-    </>
+    <ImageContainer shape={props.shape} width={props.width}>
+      <StyledImage src={src} {...props} />
+    </ImageContainer>
   );
 };
+
 Image.defaultProps = {
-  src: `https://picsum.photos/100/200`,
```
Comment (alirezatahani0) [inline src/components/image/content/Image.tsx:13]: plz add default pic 

https://us-wbe-img.gr-cdn.com/user/8a9c5bd8-0d48-4f74-839c-04b16e7e404a/610efd88-ae0d-48b7-96fb-8bc45cc8ad61.png

if error hapened on loading image or src is null

File: src/global/Global.ts [hunk 6]
```diff
@@ -201,10 +201,13 @@ const theme = {
       },
     },
   },
-
   general: {
     borderRadius: 50,
     transition: 'all 100ms ease-in',
+    imageShapes: {
+      rounded: { borderRadius: 6 },
+      circle: { borderRadius: '90%' },
```
Comment (alirezatahani0) [inline src/global/Global.ts]: circle 50 % ??

**Review by alirezatahani0 — CHANGES_REQUESTED** [review]

File: src/global/Global.ts (hunk 6 above)
Comment (Amirjf) [reply src/global/Global.ts]: ok

File: src/components/image/content/Image.tsx (hunk 5 above)
Comment (Amirjf) [reply src/components/image/content/Image.tsx:13]: bayad sharte khassi bezaram ya default bezaram faghat okeye ??

**Pull request merged**